openff/toolkit/topology/topology.py

```diff
@@ -1417,12 +1421,12 @@ def from_openmm(
                 )
                 if isomorphic:
                     # Take the first valid atom indexing map
-                    first_topology_atom_index = min(mapping.keys())
+                    first_topology_atom_index = min(mapping.keys())  # type: ignore[union-attr]
                     topology_molecules_to_add.append(
                         (
                             first_topology_atom_index,
                             unq_mol_G,
-                            mapping.items(),
+                            mapping.items(),  # type: ignore[union-attr]
```

Comment:
These could be fixed by overloading `Molecule.are_isomorphic` with an absolute mess of gore that I'm not inclined to suggest (or an API break to return an empty dictionary instead of `None`):


```python
    @overload
    @staticmethod
    def are_isomorphic(
        mol1: Union["FrozenMolecule", "_SimpleMolecule", nx.Graph],
        mol2: Union["FrozenMolecule", "_SimpleMolecule", nx.Graph],
        return_atom_map: Literal[False] = False,
        aromatic_matching: bool = True,
        formal_charge_matching: bool = True,
        bond_order_matching: bool = True,
        atom_stereochemistry_matching: bool = True,
        bond_stereochemistry_matching: bool = True,
        strip_pyrimidal_n_atom_stereo: bool = True,
        toolkit_registry: TKR = GLOBAL_TOOLKIT_REGISTRY,
    ) -> Tuple[bool, None]:
        ...

    @overload
    @staticmethod
    def are_isomorphic(
        mol1: Union["FrozenMolecule", "_SimpleMolecule", nx.Graph],
        mol2: Union["FrozenMolecule", "_SimpleMolecule", nx.Graph],
        return_atom_map: Literal[True] = False,
        aromatic_matching: bool = True,
        formal_charge_matching: bool = True,
        bond_order_matching: bool = True,
        atom_stereochemistry_matching: bool = True,
        bond_stereochemistry_matching: bool = True,
        strip_pyrimidal_n_atom_stereo: bool = True,
        toolkit_registry: TKR = GLOBAL_TOOLKIT_REGISTRY,
    ) -> Tuple[bool, Dict[int, int]]:
        ...

    @staticmethod
    def are_isomorphic(
        mol1: Union["FrozenMolecule", "_SimpleMolecule", nx.Graph],
        mol2: Union["FrozenMolecule", "_SimpleMolecule", nx.Graph],
        return_atom_map: bool = False,
        aromatic_matching: bool = True,
        formal_charge_matching: bool = True,
        bond_order_matching: bool = True,
        atom_stereochemistry_matching: bool = True,
        bond_stereochemistry_matching: bool = True,
        strip_pyrimidal_n_atom_stereo: bool = True,
        toolkit_registry: TKR = GLOBAL_TOOLKIT_REGISTRY,
    ) -> Tuple[bool, Optional[Dict[int, int]]]:
        # Okay, now you're allowed to do the implementation :)

Reply:
Gore indeed! I'll be keeping the `@overloads` on the outside of the airlock for now.